src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,505 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2018 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+
+/**
+ * @author l.casciaro
+ *
+ */
+public class TestNoSQLInjection extends AbstractAppParamPlugin {
+	
+	
+	
+	// Prefix for internationalised messages used by this rule
+	 
+	private static final String MESSAGE_PREFIX = "ascanalpha.testnosqlinjection.";
+	private static final String JSON_ATTACK = "json";
+	private static final String PARVAL_ATTACK = "paramvalue";
+	private static final String VALUE_ATTACK = "value";
+	private static final String CRASH_ATTACK = "crash";
+	private static final String TIMED_ATTACK ="timed";
+	private static final String TOKEN ="OWASP_ZAP_TOKEN_INJECTION";
+	private static final int SLEEP_TIME_SHORT=500;
+	private static final int SLEEP_TIME_LONG=2000;
+	private static final int DELTA_TIME=200;
+
+	private static final int ID = 40033;
+	private static final int CWED_ID = 943;
+	private static final int WASC_ID = 19;
+	
+	private static final String MONGO_INDEX = "mongo";
+	private static final String COUCH_INDEX = "couch";
+	private String indexDbSel = MONGO_INDEX;
+	
+	
+	
+	//START MongoDB
+	private static final String[] MONGO_ERROR_MATCHING_STRING = 
+
+			new String[] { 
+					"retval", "Unexpected token", "mongoDB", "SyntaxError", "ReferenceError", "Illegal", TOKEN };

```

Comment:
Breaking this across multiple lines seems unnecessary unless it's meant to be formatted differently.